libs/portals/admin/ids-admin/src/components/forms/AddPermissions/AddPermissions.tsx

```diff
@@ -8,125 +8,138 @@ import {
 } from '@island.is/island-ui/core'
 import { m } from '../../../lib/messages'
 import { useLocale } from '@island.is/localization'
-import React from 'react'
-import { Form } from 'react-router-dom'
+import React, { useContext } from 'react'
 import { ShadowBox } from '../../ShadowBox/ShadowBox'
+import { ClientContext } from '../../../shared/context/ClientContext'
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import { getTranslatedValue } from '@island.is/portals/core'
 
 interface AddPermissionsProps {
   isVisible: boolean
-
-  onClose(): void
+  onClose: () => void
+  onAdd: (permissions: AuthAdminClientAllowedScope[]) => void
 }
 
-type Permission = {
-  id: string
-  label: string
-  description: string
-  api: string
-  locked?: boolean
-}
+function AddPermissions({ isVisible, onClose, onAdd }: AddPermissionsProps) {
```

Comment:
When this modal starts quering the scopes available in the tenant remember to filter out allowed scopes.